clients/src/main/java/org/apache/kafka/clients/producer/internals/TransactionManager.java

```diff
@@ -790,7 +781,6 @@ private void adjustSequencesDueToFailedBatch(ProducerBatch batch) {
                 throw new IllegalStateException("Sequence number for batch with sequence " + inFlightBatch.baseSequence()
                         + " for partition " + batch.topicPartition + " is going to become negative: " + newSequence);
 
-            log.info("Resetting sequence number of batch with current sequence {} for partition {} to {}", inFlightBatch.baseSequence(), batch.topicPartition, newSequence);
```

Comment:
why should we remove this log?

Reply:
Cause `inFlightBatch.resetProducerState` will be called immediately. I think this log is duplicate in `resetProducerState` method.

Reply:
Aha, nice catch!